attitude_fw/ecl_pitch_controller.cpp

```diff
@@ -148,5 +137,14 @@ float ECL_PitchController::control_euler_rate(const struct ECL_ControlData &ctl_
 	_bodyrate_setpoint = cosf(ctl_data.roll) * _rate_setpoint +
 			     cosf(ctl_data.pitch) * sinf(ctl_data.roll) * ctl_data.yaw_rate_setpoint;
 
+	 /* limit the rate */
+ 	if (_max_rate > 0.01f && _max_rate_neg > 0.01f) {
```

Comment:
I guess it could be possible that someone wants to completely disable the limits (though probably not recommended) -- this would be the "unlimited case". And if someone simply wanted to disable the controller, they could put very small limits. Though I'm not exactly sure when disabling the rate controller in this manner would ever make sense. Maybe it's ok to leave it as it is only for the case that someone really does want unlimited rate setpoints? Unless we want to deny the user this option. But then we would need to define a default value. And then on the small limit side of things.. this is anyway the responsibility of the user to put proper values (e.g. it probably never makes sense to set a value of 0.01).

Reply:
Disabling ecl yaw if your vehicle doesn't have yaw actuation? The roll and pitch body rate transformation is a function of yaw rate setpoint.

https://github.com/PX4/ecl/blob/master/attitude_fw/ecl_roll_controller.cpp#L137
https://github.com/PX4/ecl/blob/master/attitude_fw/ecl_pitch_controller.cpp#L149

Reply:
Wouldnt this be the job of the mixer though? In the rate controllers, we are still abstracted to setting some normalized angular acceleration command, then the mixer is suppose to interpret this for whatever actuation method. So yaw rate tracking can still be possible even without e.g. a rudder, and should probably still be considered. In the case someone really doesnt want turn coordination, they could also set the gains to zero, or simply dont mix that channel into the actuators.

Reply:
Yes that makes sense, but I'm questioning the value of pursuing a coordinated turn in the first place when the controller is oblivious to the fact that's it's missing an axis. Which is less wrong?

Regardless, why leave the hole at < 0.01f? It's less work and more intuitive to let 0 mean 0 instead of unlimited. It's easy to set an effectively unlimited value, but a user (or even developer) has no way of knowing the difference between an incredibly small limit and unlimited being right next to each other.

Reply:
Sure. Agreed. I'll take it out.